qa/workunits/fs/fscrypt.sh

```diff
@@ -2,118 +2,84 @@
 
 set -xe
 
-mydir=`dirname $0`
+mydir=$(dirname "$0")
 
-if [ $# -ne 2 ]
-then
-	echo "2 parameters are required!\n"
-	echo "Usage:"
-	echo "  fscrypt.sh <type> <testdir>"
-	echo "  type: should be any of 'none', 'unlocked' or 'locked'"
-	echo "  testdir: the test direcotry name"
-	exit 1
-fi
-
-fscrypt=$1
-testcase=$2
-testdir=fscrypt_test_${fscrypt}_${testcase}
-mkdir $testdir
-
-XFSPROGS_DIR='xfprogs-dev-dir'
-XFSTESTS_DIR='xfstest-dev-dir'
-export XFS_IO_PROG="$(type -P xfs_io)"
-
-# Setup the xfstests env
-setup_xfstests_env()
-{
-	git clone https://git.ceph.com/xfstests-dev.git $XFSTESTS_DIR --depth 1
-	pushd $XFSTESTS_DIR
-	. common/encrypt
-	popd
-}
-
-install_deps()
-{
-	local system_value=$(sudo lsb_release -is | awk '{print tolower($0)}')
-	case $system_value in
-		"centos" | "centosstream" | "fedora")
-			sudo yum install -y inih-devel userspace-rcu-devel \
-				libblkid-devel gettext libedit-devel \
-				libattr-devel device-mapper-devel libicu-devel
-			;;
-		"ubuntu" | "debian")
-			sudo apt-get install -y libinih-dev liburcu-dev \
-				libblkid-dev gettext libedit-dev libattr1-dev \
-				libdevmapper-dev libicu-dev pkg-config
-			;;
-		*)
-			echo "Unsupported distro $system_value"
-			exit 1
-			;;
-	esac
-}
-
-# Install xfsprogs-dev from source to support "add_enckey" for xfs_io
-install_xfsprogs()
-{
-	local install_xfsprogs=0
+# Assumes fscrypt_cli_setup.sh has been called by the YAML file used to execute this script
 
-	xfs_io -c "help add_enckey" | grep -q 'not found' && install_xfsprogs=1
-
-	if [ $install_xfsprogs -eq 1 ]; then
-		install_deps
+if [ $# -ne 2 ]; then
+    echo "2 parameters are required!"
+    echo "Usage:"
+    echo "  fscrypt.sh <none|unlocked|locked> <testcase>"
+    echo "  testcase: the test workunit name"
+    exit 1
+fi
 
-		git clone https://git.ceph.com/xfsprogs-dev.git $XFSPROGS_DIR --depth 1
-		pushd $XFSPROGS_DIR
-		make
-		sudo make install
-		popd
-	fi
-}
+fscrypt_type=$1
+test_case=$2
+test_dir=fscrypt_test_${fscrypt_type}_${test_case}
 
-clean_up()
-{
-	rm -rf $XFSPROGS_DIR
-	rm -rf $XFSTESTS_DIR
-	rm -rf $testdir
-}
+# Ensure fscrypt CLI is installed and in PATH
+FSCRYPT_CLI="$(type -P fscrypt)"
+if [ -z "$FSCRYPT_CLI" ]; then
+    echo "fscrypt CLI not found after setup. Ensure it was installed correctly."
+    exit 1
+fi
 
-# For now will test the V2 encryption policy only as the
-# V1 encryption policy is deprecated
+# Initialize global fscrypt config
+sudo $FSCRYPT_CLI setup --force --verbose --all-users
 
-install_xfsprogs
-setup_xfstests_env
+# can't think of better way of getting ceph mount name... leaving it hardcoded, temporary
```

Comment:
I believe the cwd on start should be at the mount point. The previous version of fscrypt.sh uses relative paths so that's why I think cwd would work.

Reply:
The `pwd` command returns `/home/ubuntu/cephtest/mnt.0/client.0/tmp` while the mount point is at `/home/ubuntu/cephtest/mnt.0`. Of course I can take it from the PWD by removing the last 2 names in the path, but I guess it's still fragile. I didn't see any ceph cli command to get the mount point info either.